pageserver/src/tenant/timeline.rs

```diff
@@ -3543,32 +3559,10 @@ impl Timeline {
 
         // This iterator walks through all key-value pairs from all the layers
         // we're compacting, in key, LSN order.
-        let all_values_iter = itertools::process_results(
-            deltas_to_compact
-                .iter()
-                .map(|l| -> Result<_> { Ok(l.load_val_refs(ctx)?.into_iter()) }),
-            |iter_iter| {
-                iter_iter.kmerge_by(|a, b| {
-                    let (a_key, a_lsn, _) = a;
-                    let (b_key, b_lsn, _) = b;
-                    (a_key, a_lsn) < (b_key, b_lsn)
-                })
-            },
-        )?;
+        let all_values_iter = all_value_refs.into_iter();
 
         // This iterator walks through all keys and is needed to calculate size used by each key
-        let mut all_keys_iter = itertools::process_results(
-            deltas_to_compact
-                .iter()
-                .map(|l| -> Result<_> { Ok(l.load_keys(ctx)?.into_iter()) }),
-            |iter_iter| {
-                iter_iter.kmerge_by(|a, b| {
-                    let (a_key, a_lsn, _) = a;
-                    let (b_key, b_lsn, _) = b;
-                    (a_key, a_lsn) < (b_key, b_lsn)
-                })
-            },
-        )?;
+        let mut all_keys_iter = all_keys.into_iter();
 
         stats.prepare_iterators_micros = stats.read_lock_drop_micros.till_now();
```

Comment:
This probably needs to be moved downwards if it is measuring the lifetime of PageReadGuard. Or maybe it wasn't doing that.

Reply:
Yeah ideally one would also rename the metric. `prepare_iterators_micros` is not accurate any more, as the cost is not in the preparation of the iterators, but in the preparation of the vecs.

Reply:
I have addressed this in #4888.